packages/core/types/src/modules/documents/service-instance.ts

```diff
@@ -1,76 +1,145 @@
-import type { Utils } from '../..';
-
+import type * as Utils from '../../utils';
 import type * as UID from '../../uid';
 
 import type { IsDraftAndPublishEnabled } from './draft-and-publish';
 import type * as Params from './params/document-engine';
 import type * as Result from './result/document-engine';
 
+export type ServiceInstance<TContentTypeUID extends UID.ContentType = UID.ContentType> = {
+  findMany<const TParams extends Params.FindMany<TContentTypeUID>>(
+    params?: TParams
+  ): Result.FindMany<TContentTypeUID, TParams>;
+
+  findFirst<const TParams extends Params.FindFirst<TContentTypeUID>>(
+    params?: TParams
+  ): Result.FindFirst<TContentTypeUID, TParams>;
+
+  findOne<const TParams extends Params.FindOne<TContentTypeUID>>(
+    params: TParams
+  ): Result.FindOne<TContentTypeUID, TParams>;
+
+  delete<const TParams extends Params.Delete<TContentTypeUID>>(
+    params: TParams
+  ): Result.Delete<TContentTypeUID, TParams>;
+
+  create<const TParams extends Params.Create<TContentTypeUID>>(
+    params: TParams
+  ): Result.Create<TContentTypeUID, TParams>;
+
+  update<const TParams extends Params.Update<TContentTypeUID>>(
+    params: TParams
+  ): Result.Update<TContentTypeUID, TParams>;
+
+  count<const TParams extends Params.Count<TContentTypeUID>>(params: TParams): Result.Count;
+
+  /**
+   * @internal
```

Comment:
Can you also activate the skipInternal during build ?

Reply:
we pbly need to figure a way to use the src code instead of the built types locally. we can do that later 👍